[bug-fix] squash: validate type of sqashed fields also

grrtrr opened the pull request:
If the user adds a 'squash' tag to a non-struct type, the call to
structType.NumField() will silently fail, the application stops.

Fixed by adding an error if the squashed field's type is not struct.

mitchellh commented:
Sorry I never saw this. This looks good but needs a unit test to cover it. If you get around to it I can merge quickly. Otherwise when I have time I'll add it and merge.

grrtrr commented:
Thank you for getting back. I rebased and added a unit test. Without the patch, the unit test produces the following panic: 
```
panic: reflect: NumField of non-struct type [recovered]
	panic: reflect: NumField of non-struct type
```

mitchellh commented:
Thanks, perfect!
